Test.hs

```diff
@@ -10,7 +10,7 @@ import qualified Pangraph.VHDL.EnvironmentWriter  as VHDL_E
 import qualified Pangraph.VHDL.GraphWriter        as VHDL_G
 
 
-type Parser = GraphML_P.Template -> BS.ByteString -> Either BS.ByteString P.Pangraph
+type Parser = BS.ByteString -> (Maybe P.Pangraph)
```

Comment:
I don't see a need for this type synonym. Looks like it's used only once in the code, and in that place I'd prefer to actually see the full signature, since `parser :: Parser` doesn't tell me anything :-)